src/asound/bluealsa-pcm.c

```diff
@@ -770,11 +831,17 @@ static int bluealsa_pause(snd_pcm_ioplug_t *io, int enable) {
 		 * the server will not be paused while we are processing a transfer. */
 		pthread_mutex_lock(&pcm->mutex);
 		pcm->pause_state |= BA_PAUSE_STATE_PENDING;
-		while (!(pcm->pause_state & BA_PAUSE_STATE_PAUSED) && pcm->ba_pcm_fd != -1)
+		while (!(pcm->pause_state & BA_PAUSE_STATE_PAUSED) && pcm->connected)
 			pthread_cond_wait(&pcm->pause_cond, &pcm->mutex);
 		pthread_mutex_unlock(&pcm->mutex);
 	}
 
+	if (!pcm->connected) {
+		pcm->pause_state = BA_PAUSE_STATE_RUNNING;
```

Comment:
Do we need to set the `pause_state` in case of error? If so, it will have to be done under mutex, otherwise TSAN might report race warning (this flag is set in the IO thread as well).